test/e2e/sync/sync.go

```diff
@@ -0,0 +1,449 @@
+package sync
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/controllerv2"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+	"github.com/ppc64le-cloud/pvsadm/test/e2e/framework"
+	"gopkg.in/yaml.v2"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	APIKey      = string(os.Getenv("IBMCLOUD_API_KEY"))
+	FolderName  = "folder" + strings.ToLower(generateRandomString(6))
+	PlanSlice   = []string{"smart", "standard", "vault", "cold"}
+	RegionSlice = []string{"us-east", "jp-tok", "us-south"}
+)
+
+const (
+	ServiceType            = "cloud-object-storage"
+	ResourceGroupAPIRegion = "global"
+	ServicePlan            = "standard"
+	ResourceGrp            = "powervs-ipi-resource-group"
+	Debug                  = false
+	Recursive              = false
+	InstanceType           = "service_instance"
+	NoOfSources            = 2
+	NoOfTargetsPerSource   = 2
+	NoOfObjects            = 5
+)
+
+type Spec []struct {
+	Source `yaml:"source"`
+	Target `yaml:"target"`
+}
+
+type Source struct {
+	Bucket string `yaml:"bucket"`
+	Cos    string `yaml:"cos"`
+	Object string `yaml:"object"`
+	Plan   string `yaml:"plan"`
+	Region string `yaml:"region"`
+}
+
+type Target []struct {
+	Bucket string `yaml:"bucket"`
+	Plan   string `yaml:"plan"`
+	Region string `yaml:"region"`
+}
+
+func runSyncCMD(args ...string) (int, string, string) {
+	args = append([]string{"image", "sync"}, args...)
+	return utils.RunCMD("pvsadm", args...)
+}
+
+func randomInt(min, max int) int {
+	return min + rand.Intn(max-min)
+}
+
+func generateRandomString(length int) string {
+	rand.Seed(time.Now().UnixNano())
+	bytes := make([]byte, length)
+	for i := 0; i < length; i++ {
+		bytes[i] = byte(randomInt(65, 91))
+	}
+	return string(bytes)
+}
+
+func generateStruct() Spec {
+
+	fmt.Println("STEP: Generating Struct")
+	spec := make(Spec, NoOfSources)
+
+	for src := 0; src < NoOfSources; src++ {
+		spec[src].Source.Bucket = "bucket-" + strings.ToLower(generateRandomString(6))
+		spec[src].Source.Cos = "COS-Test-" + generateRandomString(6)
+		spec[src].Source.Object = ""
+		spec[src].Source.Plan = PlanSlice[randomInt(0, len(PlanSlice))]
+		spec[src].Source.Region = RegionSlice[randomInt(0, len(RegionSlice))]
+		spec[src].Target = make(Target, NoOfTargetsPerSource)
+
+		for tgt := 0; tgt < NoOfTargetsPerSource; tgt++ {
+			spec[src].Target[tgt].Bucket = "bucket-" + strings.ToLower(generateRandomString(6))
+			spec[src].Target[tgt].Plan = PlanSlice[randomInt(0, len(PlanSlice))]
+			spec[src].Target[tgt].Region = RegionSlice[randomInt(0, len(RegionSlice))]
+		}
+	}
+	return spec
+}
+
+func createSpecFile(spec Spec) bool {
+
+	fmt.Println("STEP: Creating Spec file")
+
+	f, err := os.Create("spec.yaml")
```

Comment:
use the temdir logic for creating the directory and add defer for deleting the same.

Reply:
Tempdir logic was added to create spec file as well as for the files copied from local system to Buckets.